lib/rspec/core/did_you_mean.rb

```diff
@@ -0,0 +1,44 @@
+module RSpec
+  module Core
+    # Service object to provide did_you_mean suggestions
+    class DidYouMean
+      attr_reader :relative_file_name
+
+      def initialize(relative_file_name)
+        @relative_file_name = relative_file_name
+      end
+
+      if defined?(::DidYouMean::SpellChecker)
+        # provide probable suggestions if a LoadError
+        def call
+          checker = ::DidYouMean::SpellChecker.new(:dictionary => Dir["spec/**/*.rb"])
+          probables = checker.correct(relative_file_name)
+          return unless probables.any?
+
+          formats probables
+        end
+      else # ruby 2.3.2 or less
+        # return nil if API for ::DidYouMean::SpellChecker not supported
```

Comment:
I think this should return an empty string, or maybe even a warning?, I feel the ruby comment is irrelevant, its if the library is available or not (which could also happens if someone uninstalls a standard gem).

Reply:
What type of warning do you think we should we use? Don't think it will add noise?

Do you consider something like:

> 'did_you_mean' standard gem is not installed. It could help you to solve the loading failure
> An error occurred while loading /folder/file.rb.

Reply:
Outputting `"\nHint: Install the `did_you_mean` gem in order to provide suggestions for similarly named files." would produce:
```
An error occurred while loading #{relative_file}.
Hint: Install the `did_you_mean` gem in order to provide suggestions for similarly named files.
```

Reply:
It seems a good idea!

Reply:
done